zilliqa/src/node.rs

```diff
@@ -651,16 +659,22 @@ impl Node {
         let block = self
             .get_block(block_number)?
             .ok_or_else(|| anyhow!("missing block: {block_number}"))?;
+        if !self
+            .consensus
+            .state()
+            .forks
+            .get(block.number())
+            .executable_blocks
+        {
+            return Err(anyhow!("State required to execute request does not exist"));
+        }
         let parent = self
             .get_block(block.parent_hash())?
             .ok_or_else(|| anyhow!("missing block: {}", block.parent_hash()))?;
         let mut state = self
             .consensus
             .state()
             .at_root(parent.state_root_hash().into());
-        if state.is_empty() {
-            return Err(anyhow!("State required to execute request does not exist"));
-        }
```

Comment:
We re-execute these txs on the state at parent, so do we not need to check that the parent's state exists

Reply:
Good point, thanks - I moved the check to `apply_transaction` instead which I think makes more sense.